endpointmanager/pkg/capabilityparser/capabilityparser_test.go

```diff
@@ -0,0 +1,416 @@
+	th.Assert(t, actual == expected, fmt.Sprintf("expected %s. received %s.", expected, actual))
+
+}
+
+func Test_GetSoftwareVersion(t *testing.T) {
+	field := "software"
+	//field2 := "name"
+
+	// basic
+
+	expected := "19.4.121.0"
+	cs, err := getDSTU2CapStat()
+	th.Assert(t, err == nil, err)
+
+	actual, err := cs.GetSoftwareVersion()
+	th.Assert(t, err == nil, err)
+	th.Assert(t, actual == expected, fmt.Sprintf("expected %s. received %s.", expected, actual))
+
+	// bad format
+
+	cs1, err := getBadFormatCapStat(cs, field)
+	th.Assert(t, err == nil, err)
+
+	_, err = cs1.GetSoftwareVersion()
+	th.Assert(t, err != nil, "expected error due to bad format")
+
+	// missing field
+
+	expected = ""
+
+	cs2, err := deleteFieldFromCapStat(cs, field)
+	th.Assert(t, err == nil, err)
+
+	actual, err = cs2.GetSoftwareVersion()
+	th.Assert(t, err == nil, err)
+	th.Assert(t, actual == expected, fmt.Sprintf("expected %s. received %s.", expected, actual))
+
+	// nested bad format
+
+	cs3, err := getNestedBadFormatCapStat(cs, field, "version")
+	th.Assert(t, err == nil, err)
+
+	_, err = cs3.GetSoftwareVersion()
+	th.Assert(t, err != nil, "expected error due to bad format")
+
+	// missing nested field
+
+	expected = ""
+
+	cs4, err := deleteNestedFieldFromCapStat(cs, field, "version")
+	th.Assert(t, err == nil, err)
+
+	actual, err = cs4.GetSoftwareVersion()
+	th.Assert(t, err == nil, err)
+	th.Assert(t, actual == expected, fmt.Sprintf("expected %s. received %s.", expected, actual))
+}
+
+func Test_Equal(t *testing.T) {
+	var cs1 CapabilityStatement
+	var cs2 CapabilityStatement
+	var equal bool
+	var err error
+
+	cs1, err = getDSTU2CapStat()
+	th.Assert(t, err == nil, err)
+
+	// test nil
+	cs2, err = NewCapabilityStatement(nil)
+	th.Assert(t, err == nil, err)
+
+	equal = cs1.Equal(cs2)
+	th.Assert(t, !equal, "expected equality comparison to nil to be false")
+
+	// test equal
+	cs2, err = getDSTU2CapStat()
+	th.Assert(t, err == nil, err)
+
+	equal = cs1.Equal(cs2)
+	th.Assert(t, equal, "expected equality comparison of equal capability statement to be true")
+
+	// test not equal
+	cs2, err = deleteFieldFromCapStat(cs2, "publisher")
+	th.Assert(t, err == nil, err)
+
+	equal = cs1.Equal(cs2)
+	th.Assert(t, !equal, "expected equality comparison of unequal capability statement to be false")
+
+}
+
+func getDSTU2CapStat() (CapabilityStatement, error) {
+	path := filepath.Join("../testdata", "allscripts_capability_dstu2.json")
+	csJSON, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	cs, err := NewCapabilityStatement(csJSON)
+	if err != nil {
+		return nil, err
+	}
+	return cs, nil
+}
+
+func getBadFormatCapStat(cs CapabilityStatement, field string) (CapabilityStatement, error) {
+	var csInt map[string]interface{}
+
+	csJSON, err := cs.GetJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(csJSON, &csInt)
+	if err != nil {
+		return nil, err
+	}
+
+	csInt[field] = []int{1, 2, 3} // bad format for publisher
```

Comment:
It seems like this comment should be updated here and down below (line 357) since the field doesn't have to be "publisher"

Reply:
addressed